### PR TITLE
Des 715 Set image alt text using context form

### PR DESCRIPTION
**Pull request by lostkeys**
This demo shows how to use context forms to set the alt text of an image "inline".

### CODE REVIEW

File: images/image-alt-text-context-form.html [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+                editor.ui.registry.addContextForm('alttext', {
+                    // Define the button used in our context toolbar to open the context form
+                    launch: {
+                        type: 'contextformbutton',
+                        icon: 'alt-text',
+                        tooltip: 'Image description (Alt text)'
+                    },
+
+                    // The input field label (Currently does nothing).
+                    label: 'Image description (alt text)',
```
Comment (lostkeys) [inline images/image-alt-text-context-form.html:85]: The label property doesn't do anything what I can see. If that's the case, I'll write up a change that the label will show up as a placeholder on the input field.

File: images/image-alt-text-context-form.html [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+
+                        // Return the the alt text to be displayed as initial value
+                        return altText;
+                    },
+                    // Define the buttons associated with the context form, in this case
+                    // a save and cancel button
+                    commands: [
+                        {
+                            type: 'contextformbutton',
+                            primary: true,
```
Comment (lostkeys) [inline images/image-alt-text-context-form.html:106]: Why doesn't this close the context form? I expected it to execute the `onAction` method but it seems not to do that. Or at least it ignores the hide action

File: images/image-alt-text-context-form.html [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+        main {
+            margin: 2rem auto;
+            max-width: 780px;
+        }
+    </style>
+</head>
+<body>
+<main>
+    <textarea id="editor">
+        <h1>Yosemite</h1>
```
Comment (ashfordneil) [inline images/image-alt-text-context-form.html:158]: Is it valid HTML to put these elements inside a textarea? Would we be better off using a div?

**Review by ashfordneil — APPROVED** [review]

File: images/image-alt-text-context-form.html (hunk 3 above)
Comment (lnewson) [reply images/image-alt-text-context-form.html:158]: It's technically plain text, however, it really should technically be encoded correctly as the following, but the browser will handle that normally. We do see issues from people not encoding this correctly from time to time and it's normally in code samples.
```
<textarea id="editor">
  &lt;h1&gt;Yosemite&lt;/h1&gt;
  &lt;p&gt;Lorem ipsum dolor sit amet, consectetur adipiscing elit. Vestibulum a molestie nisl, at sodales diam. Vivamus ullamcorper convallis odio, ut lacinia sem.&lt;/p&gt;
  &lt;img src=&quot;images/yosemite.jpeg&quot; alt=&quot;A winding road with Yosemite in focus&quot;/&gt;
</textarea>
```

File: images/image-alt-text-context-form.html (hunk 1 above)
Comment (lnewson) [reply images/image-alt-text-context-form.html:85]: It's used for ARIA purposes, as it adds the `aria-label` attribute:
image

File: images/image-alt-text-context-form.html [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+                                    // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.selection/#getnode
+                                    const node = editor.selection.getNode();
+
+                                    // Set the alt attribute on the selected image
+                                    // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.domutils/#setattrib
+                                    const alt = tinymce.DOM.setAttrib(node, 'alt', contextFormApi.getValue());
+
+                                    // Close the context form
+                                    // https://www.tiny.cloud/docs/ui-components/contextform/#formapi
+                                    contextform.hide();
```
Comment (lnewson) [inline images/image-alt-text-context-form.html:124]: Typo here, as this should be `contextFormApi` and was throwing an exception. As such, that's likely why it wasn't working in your comment above.

File: images/image-alt-text-context-form.html [hunk 5]
```diff
@@ -0,0 +1,164 @@
...
+                                });
+                            }
+                        },
+                        {
+                            type: 'contextformbutton',
+                            tooltip: 'Cancel',
+                            icon: 'close',
+                            onAction: (contextform) => {
```
Comment (lnewson) [inline images/image-alt-text-context-form.html:132]: Why the different names here? As above you use `contextFormApi`, but here it's `contextform`.

**Review by lnewson — APPROVED** [review]

File: images/image-alt-text-context-form.html (hunk 3 above)
Comment (lostkeys) [reply images/image-alt-text-context-form.html:158]: We do this in all our demos in the docs so we'll continue doing this here too until there is a joint effort adressing this everywhere.

File: images/image-alt-text-context-form.html (hunk 1 above)
Comment (lostkeys) [reply images/image-alt-text-context-form.html:85]: Nice!

File: images/image-alt-text-context-form.html (hunk 4 above)
Comment (lostkeys) [reply images/image-alt-text-context-form.html:124]: Facepalm

File: images/image-alt-text-context-form.html (hunk 5 above)
Comment (lostkeys) [reply images/image-alt-text-context-form.html:132]: Imagine when computer will understand the human intention...

File: images/image-alt-text-context-form.html (hunk 2 above)
Comment (lostkeys) [reply images/image-alt-text-context-form.html:106]: Caused by not naming parameters correctly

**Pull request merged**